### DISCUSSION TITLE
Creating new custom blocks to reuse

### DISCUSSION THREAD

**Discussion by planetahuevo** [voidlabs/versafix-template#45] [Q&A]
I read this one https://github.com/voidlabs/mosaico/issues/549 and that redirected me to this repository. 
But I cannot find any docs or instructions on what is what we need to modify to be able to create our own blocks to reuse on multiple future templates. 

Can you please link or expand the instructions? 
Will this be compatible with using the Docker installation of mosaico? 
And last, 
Would be possible to share those blocks back here so they are added to the project? 

Thank you!

**Comment by bago** [comment]
First of all: creating or altering a master template is not an easy task and requires dev skills.

You can find some docs about the template language for Mosaico, in Mosaico wiki, here:
https://github.com/voidlabs/mosaico/wiki/Template-language

This specific template, versafix-template, uses one more tool to make it simpler (for us) to maintain the template:
https://github.com/voidlabs/htmml/blob/master/README.md

> Can you please link or expand the instructions?

That's what we have now: contributions are welcome.

> Will this be compatible with using the Docker installation of mosaico?

You probably need to fork the docker source to include your own master template.

> Would be possible to share those blocks back here so they are added to the project?

Well, of course, but you have to create those blocks, first, and then you can share them here and we'll evaluate if they are useful in the versafix-template template or not. We carefully chose which block to add there to find the best balance between flexibility and usability.

**Comment by planetahuevo** [comment]
Thank you for the links. I have work to do then. :)
